package.json

```diff
@@ -126,16 +127,16 @@
     "lint:staged": "lint-staged",
     "start": "wp-scripts start",
     "test": "npm-run-all --parallel test:*",
-    "test:e2e": "env-cmd --silent wp-scripts test-e2e --config=tests/e2e/jest.config.js",
+    "test:e2e": "wp-scripts test-e2e --config=tests/e2e/jest.config.js",
     "test:e2e:help": "npm run test:e2e -- --help",
     "test:e2e:watch": "npm run test:e2e -- --watch",
     "test:e2e:interactive": "npm run test:e2e -- --puppeteer-interactive",
-    "test:e2e:ci": "cross-env WP_BASE_URL=http://localhost:8890 wp-scripts test-e2e --config=tests/e2e/jest-ci.config.js --runInBand",
+    "test:e2e:ci": "npm run test:e2e -- --runInBand",
     "test:js": "wp-scripts test-unit-js --config=tests/js/jest.config.js",
     "test:js:help": "wp-scripts test-unit-js --help",
     "test:js:watch": "npm run test:js -- --watch",
-    "test:php": "vendor/bin/phpunit",
-    "test:php:help": "npm run test:php -- --help",
+    "test:php": "wp-env run phpunit 'WORDPRESS_TABLE_PREFIX=wptests_ /var/www/html/wp-content/plugins/amp/vendor/bin/phpunit -c /var/www/html/wp-content/plugins/amp/phpunit.xml.dist $npm_config_args'",
```

Comment:
I would suggest using the `tests-wordpress` container instead of the `phpunit`, because there is no xdebug support in the phpunit container.

Reply:
I've used Gutenberg setup as a base for this:

https://github.com/WordPress/gutenberg/blob/a80da699942ae2c695535cc23048d5883e6c38c1/package.json#L283

Do you think we need xdebug when running phpunit tests?

Reply:
It's not always necessary. But having xdebug during TDD is very helpful.

Reply:
xdebug is needed for code coverage, no?

Reply:
I think that xdebug will be enabled on all environments since we're start it using `wp-env start --xdebug`.

I got an error on my local when I tried one of the following:

```diff
- "test:php": "wp-env run phpunit ..."
+ "test:php": "wp-env run tests-wordpress ..."
# or
- "test:php": "wp-env run phpunit ..."
+ "test:php": "wp-env run tests-cli ..."
```

@ediamin Is there a way to check if xdebug is enabled with the `test:php` script as it is currently?

Reply:
What kind of error your're having in your local? I think by default, the xdebug is enabled only in dev env and only for `wordpress` container. Check out [this code reference](https://github.com/WordPress/gutenberg/blob/trunk/packages/env/lib/init-config.js#L87). But if we use **same PHP version** for both dev and test env, then it'll enable for both env but only for `wordpress` and `tests-wordpress` containers, no xdebug in phpunit, or cli containers.

Reply:
I'm getting the following error when running with `tests-wordpress` or `tests-cli`:

```sh
Creating 690d03eee1d37888267bcce48072f486_tests-wordpress_run ... done
/usr/local/bin/docker-entrypoint.sh: line 99: exec: WORDPRESS_TABLE_PREFIX=wptests_: not found
ERROR: 127
✖ Command failed with exit code 127
```

Reply:
Both tests-wordpress and wordpress containers have the `docker-entrypoint.sh` and other containers don't. This entrypoint contains `exec "$@"` at the end of the file. That is why when we use `...'WORDPRESS_TABLE_PREFIX=wptests_ /var/...` in npm script, it raises this error. So, we can just remove this from the script. But after that you'll have another error that says, `Error: wp-tests-config.php is missing!` because `WP_PHPUNIT__TESTS_CONFIG` env value is set only for the phpunit container in docker-compose.yml. I don't think we can fix this from the .wp-env.json settings. Instead we can set config in our bootstrap.php like following,
```
if ( ! defined( 'WP_TESTS_CONFIG_FILE_PATH' ) ) {
	define( 'WP_TESTS_CONFIG_FILE_PATH', '/var/www/html/phpunit-wp-config.php' );
}
```

Reply:
> That is why when we use `...'WORDPRESS_TABLE_PREFIX=wptests_ /var/...` in npm script, it raises this error. So, we can just remove this from the script.

Does it mean that phpunit would use the default `wp_` table prefix? If yes, then we'll have an issue with phpunit and e2e tests interfering with each other.

Reply:
@ediamin If it's possible to use distinct table prefixes, go ahead and push the required changes to this branch.

Reply:
I think we can use `env VAR_NAME=var_value` to solve this problem in the script,

```
"test:php": "wp-env run tests-wordpress 'env WORDPRESS_TABLE_PREFIX=wptests_ WP_PHPUNIT__TESTS_CONFIG=/var/www/html/phpunit-wp-config.php /var/www/html/wp-content/plugins/amp/vendor/bin/phpunit -c /var/www/html/wp-content/plugins/amp/phpunit.xml.dist $npm_config_args'",
```

Reply:
@ediamin This works well for changing the table prefix however now I'm getting some xdebug issues:

```sh
[12-Jan-2022 14:14:14 UTC] Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9003 (through xdebug.client_host/xdebug.client_port) :-(
[12-Jan-2022 14:14:14 UTC] Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9003 (through xdebug.client_host/xdebug.client_port) :-(
Installing...
[12-Jan-2022 14:14:15 UTC] Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9003 (through xdebug.client_host/xdebug.client_port) :-(
[12-Jan-2022 14:14:15 UTC] Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9003 (through xdebug.client_host/xdebug.client_port) :-(
[12-Jan-2022 14:14:15 UTC] Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9003 (through xdebug.client_host/xdebug.client_port) :-(
```

Despite the notices, the test suites are executed as expected.

How about adding a new task for running phpunit with xdebug like:

```suggestion
    "test:php": "wp-env run phpunit 'WORDPRESS_TABLE_PREFIX=wptests_ /var/www/html/wp-content/plugins/amp/vendor/bin/phpunit -c /var/www/html/wp-content/plugins/amp/phpunit.xml.dist $npm_config_args'",
    "test:php:xdebug": "wp-env run tests-wordpress 'env WORDPRESS_TABLE_PREFIX=wptests_ WP_PHPUNIT__TESTS_CONFIG=/var/www/html/phpunit-wp-config.php /var/www/html/wp-content/plugins/amp/vendor/bin/phpunit -c /var/www/html/wp-content/plugins/amp/phpunit.xml.dist $npm_config_args'",
```

Reply:
Sure, we can have a separate task for that. Actually if you start debugging or start listening for the xdebug in your IDE/editor then those message will be gone.

Reply:
That's correct, thanks! After I turned on listening for xdebug connections, I got no errors and debugging within phpunit tests started to work as expected.

I think we shouldn't expect every contributor to turn on xdebug before running the tests. That's why having a separate script for running phpunit _with_ xdebug support makes even more sense (added in 2beedcb).

Reply:
I've incorporated this into https://github.com/GoogleChromeLabs/pwa-wp/pull/778